helloworld/greet/BUILD

```diff
@@ -3,13 +3,19 @@
 
 # This target sets the metadata for all the Python non-test files in this directory.
 #
-# * `name` defaults to the name of this directory, i.e., `greet`.
-# * `sources` defaults to ['*.py', '!*_test.py', '!test_*.py', '!conftest.py'].
-# * `dependencies` are inferred.
-python_library()
+# * Pants cannot infer dependencies on `resources` targets, so we explicitly add the dep.
+python_library(
+    name="lib",
+    dependencies=[":translations"],
+)
 
 # This target sets the metadata for all the Python test files in this directory.
-#
-# * `sources` defaults to ['*_test.py', 'test_*.py', 'conftest.py'].
-# * `dependencies` are inferred.
-python_tests(name='tests')
+python_tests(
+    name="tests",
+)
+
+# This target teaches Pants about our JSON file, which allows for other targets to depend on it.
+resources(
+    name="translations",
+    sources=["translations.json"],
```

Comment:
Maybe glob `*.json`? A bit shorter, and less fragile.

Reply:
In 2.8, I will update this to use `resource` instead of `resources`